docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,455 @@
+* Modifying the  default gateway and management addresses on a Layer2 UDN should not be possible after the network
+was created.
+
+The scenarios mentioned above have to cover both IPv4 and IPv6 IP families.
+
+
+### Documentation Details
+
+## Risks, Known Limitations and Mitigations
+
+* Modifying the 'v1.multus-cni.io/default-network' value after the pod was created could have unpredictable
+consequences.
+To mitigate this introduce a Validating Admission Policy described in [Implementation Details](#implementation-details).
+
+* By allowing users to specify the IP and MAC addresses for the pods there is a risk of conflicts.
+To mitigate this OVN-Kubernetes will check that the requested addresses are not currently used in the UDN.
+There is still a risk that the user picks an address that's consumed by something outside of the UDN but that's beyond
+what OVN-Kubernetes controls and can check.
+
+* The dynamic, per-node subnet allocation in Layer3 UDNs, where each node has a unique default gateway and
+management IP, makes user-specified UDN gateway/management IPs and static pod IP/MAC assignments very complex. This
+enhancement will not support Layer3 UDNs.
+
+* BGP support today is limited to cluster UDNs, to ensure a non-NATed traffic for pods with pre-configured addresses
+the user has to use a cluster UDN to configure the network. This is a limitation unrelated to this enhancement
+and it is possible it will be solved in the future.
+
+* By consuming the 'v1.multus-cni.io/default-network' annotation for altering the primary UDNs pod configuration the
+user won't be able to use it for configuring the cluster default network attachment. This is acceptable as there is
+currently no support for modifying the cluster default network through this annotation while using primary UDNs.
+If there is a requirement in the future another mechanism can be considered.
+
+## OVN Kubernetes Version Skew
+
+## Alternatives
+
+* Instead of the [Pod network identity](#pod-network-identity) approach, we could expand the
+IPAMClaim API. It currently lacks IP request capabilities, and using IPAMClaim for MAC addresses
+is confusing. Introducing a new API would mean deprecating the IPAMClaim, while managing
+upgrades and supporting both solutions for a period of time. This requires significant effort, which
+is not feasible at this time.
+
+* As described in the [NetworkSelectionElement annotation](#networkselectionelement-annotation) section, using the
+`v1.multus-cni.io/default-network` annotation means Multus strictly requires this annotation's value to reference an
+existing NAD. An alternative to the proposed approach would be to reference the NAD that defines the primary network.
+It was discarded as it would require OVN-Kubernetes to modify the CNI handling logic because multus
+would target the CNI requests towards the custom network. Additionally it would require users to determine the exact
```

Comment:
hmm shouldn't the request be targeted to the custom network?

Reply:
I can't remember what I meant, I wouldn't trust any of my comments towards end of the enhancement. resolving this